app/src/components/Modals/delete-activity-modal.tsx

```diff
@@ -14,27 +14,83 @@ import {
   Box,
   Badge,
   ModalFooter,
+  useToast,
 } from "@chakra-ui/react";
 import { TFunction } from "i18next";
 import React, { FC } from "react";
 import { Trans } from "react-i18next";
+import { CheckCircleIcon } from "@chakra-ui/icons";
 
 import { FiTrash2 } from "react-icons/fi";
 
 interface DeleteAllActivitiesModalProps {
   isOpen: boolean;
   onClose: any;
   t: TFunction;
+  selectedActivityValue: ActivityValue;
+  resetSelectedActivityValue: () => void;
+  inventoryId: string;
 }
 
 const DeleteActivityModal: FC<DeleteAllActivitiesModalProps> = ({
   isOpen,
   onClose,
   t,
+  selectedActivityValue,
+  resetSelectedActivityValue,
+  inventoryId,
 }) => {
+  const toast = useToast();
+  const [deleteActivityValue, { isLoading }] = useDeleteActivityValueMutation();
+
+  // define the function to delete all activities
+  const handleDeleteActivity = async () => {
+    // call the delete all activities mutation
+    const response = await deleteActivityValue({
+      inventoryId,
+      activityValueId: selectedActivityValue.id,
+    });
+    if (response.data) {
+      // TODO create toast wrapper for success state
```

Comment:
Yes, that would be great :+1: We have this code copied quite a few times throughout the codebase, so it would be great to extract to a component that can be reused which then does the different states? So this would just become a call that takes in a status, title, maybe icon etc. and the rendering and styling is up to our new toast wrapper component.